mscore/plugin/pluginUpdater.cpp

```diff
@@ -0,0 +1,994 @@
+//=============================================================================
+//  MuseScore
+//  Music Composition & Notation
+//
+//  Copyright (C) 2019 Werner Schweer and others
+//
+//  This program is free software; you can redistribute it and/or modify
+//  it under the terms of the GNU General Public License version 2
+//  as published by the Free Software Foundation and appearing in
+//  the file LICENSE.GPL
+//=============================================================================
+
+#include "musescore.h"
+#include "preferences.h"
+#include "resourceManager.h"
+#include "downloadUtils.h"
+#include "plugin/pluginUpdater.h"
+#include "thirdparty/qzip/qzipreader_p.h"
+#include "ui_resourceManager.h"
+
+namespace Ms {
+
+static int PluginStatus_id = qRegisterMetaType<PluginStatus>("PluginStatus");
+
+static inline std::tuple<bool, bool, bool> compatFromString(const QString& raw) {
+      return std::make_tuple(raw.contains("1.x"), raw.contains("2.x"), raw.contains("3.x"));
+}
+
+//---------------------------------------------------------
+//   GitHub utility functions
+//---------------------------------------------------------
+static inline QString githubReleaseAPI(const QString &user, const QString &repo) { return "https://api.github.com/repos/" + user + "/" + repo + "/releases"; }
+static inline QString githubCommitAPI(const QString &user, const QString &repo, const QString &ref = "")
+      {
+      return "https://api.github.com/repos/" + user + "/" + repo + "/commits" + (ref.isEmpty() ? "" : ('/' + ref));
+      }
+static inline QString githubLatestArchiveURL(const QString &user, const QString &repo, const QString &branch = "master")
+      {
+      //return "https://api.github.com/repos/" + user + "/" + repo + "/zip/" + branch;
+      return "https://github.com/" + user + '/' + repo + "/archive/" + branch + ".zip";
+      }
+static QString getLatestCommitSha(const QString& user, const QString &repo, const QString &branch = "master")
+      {
+      QString api_url = githubCommitAPI(user, repo, branch);
+      DownloadUtils json;
+      json.setTarget(api_url);
+      json.download();
+      QJsonDocument json_doc = QJsonDocument::fromJson(json.returnData());
+      QJsonObject latest_commit;
+      if (json_doc.isArray())
+            // there are  multiple commits, fetch the latest
+            latest_commit = json_doc.array().first().toObject();
+      else
+            latest_commit = json_doc.object();
+      if (latest_commit.contains("sha"))
+            return latest_commit["sha"].toString();
+      else return {};
+      }
+static inline bool isCompatibleRelease(const QString& branch)
```

Comment:
really ìnline`? Seems to large and complex for that

Reply:
OK, changed to non-inline.